bundles/org.openhab.binding.sncf/src/main/java/org/openhab/binding/sncf/internal/handler/StationHandler.java

```diff
@@ -0,0 +1,257 @@
+        logger.debug("Scheduling update in {} seconds.", wishedDelay);
+        refreshJob = scheduler.schedule(() -> updateThing(), wishedDelay, TimeUnit.SECONDS);
+    }
+
+    private void updateThing() {
+        SncfBridgeHandler bridgeHandler = getBridgeHandler();
+        if (bridgeHandler != null) {
+            scheduler.submit(() -> updatePassage(bridgeHandler, GROUP_ARRIVAL));
+            scheduler.submit(() -> updatePassage(bridgeHandler, GROUP_DEPARTURE));
+        }
+    }
+
+    private void updatePassage(SncfBridgeHandler bridgeHandler, String movement) {
+        try {
+            bridgeHandler.getNextPassage(stationId, movement).ifPresentOrElse(passage -> {
+                getThing().getChannels().stream().map(Channel::getUID)
+                        .filter(channelUID -> isLinked(channelUID) && movement.equals(channelUID.getGroupId()))
+                        .forEach(channelUID -> {
+                            State state = getValue(channelUID.getIdWithoutGroup(), passage, movement);
+                            updateState(channelUID, state);
+                        });
+                StopDateTime stopTime = passage.stopDateTime;
+                scheduleRefresh(fromDTO(
+                        GROUP_DEPARTURE.equals(movement) ? stopTime.departureDateTime : stopTime.arrivalDateTime));
+            }, () -> {
+                logger.debug("No {} available", movement);
+                scheduleRefresh(ZonedDateTime.now().plusMinutes(5));
+            });
+            updateStatus(ThingStatus.ONLINE);
+        } catch (SncfException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            freeRefreshJob();
+        }
+    }
+
+    private State getValue(String channelId, Passage passage, String group) {
+        switch (channelId) {
+            case DIRECTION:
+                return fromNullableString(passage.route.direction.name);
+            case CODE:
+                return fromNullableString(passage.displayInformations.code);
+            case COMMERCIAL_MODE:
+                return fromNullableString(passage.displayInformations.commercialMode);
+            case NAME:
+                return fromNullableString(passage.displayInformations.name);
+            case NETWORK:
+                return fromNullableString(passage.displayInformations.network);
+            case TIMESTAMP:
+                return fromNullableDTO(group.equals(GROUP_ARRIVAL) ? passage.stopDateTime.arrivalDateTime
+                        : passage.stopDateTime.departureDateTime);
+        }
+        return UnDefType.NULL;
+    }
+
+    private State fromNullableString(@Nullable String aValue) {
+        return aValue != null ? StringType.valueOf(aValue) : UnDefType.NULL;
+    }
+
+    private @Nullable ZonedDateTime fromDTO(@Nullable String dateTime) {
+        return dateTime != null ? ZonedDateTime.parse(dateTime + zoneOffset, NAVITIA_DATE_FORMAT).plusMinutes(1) : null;
```

Comment:
Do you really need to add one minute ?
I will compare timestamps with my SNCF app as soon as data are available again !

Reply:
In fact, this is probably to avoid retrieving the same train with the same timestamp ?
You refresh one minute later  so that either you will switch to another train or the same train but with a new timestamp.

Reply:
You're correct. I introduced that to be sure I will not be requesting for the same train that left the train station before the updated information is available. On the other hand I did not figure it will impact channel data publications. I've got to correct his.

Reply:
Changed.